stdlib/Random/src/MersenneTwister.jl

```diff
@@ -9,9 +9,9 @@ const MT_CACHE_I = 501 << 4 # number of bytes in the UInt128 cache
 
 mutable struct MersenneTwister <: AbstractRNG
     seed::Any
```

Comment:
does a `MersenneTwister` really need to store it's seed?

Reply:
It's there to recreate the state of an instance from few values, so that what `show` outputs corresponds to a real constructor. E.g
```julia
julia> MersenneTwister("a seed")
MersenneTwister("a seed")
```
But I was thinking to perhaps normalize seeds into an `UInt128` value and store that instead of the original seed. The example above would then look like
```julia
julia> MersenneTwister("a seed")
MersenneTwister(0xb4802685c420b29be64de36a1f90815f)

julia> MersenneTwister(0xb4802685c420b29be64de36a1f90815f)
MersenneTwister(0xb4802685c420b29be64de36a1f90815f)
```
This would involve an additional round with `SeedHasher`, but I'm precisely preparing another PR to make that much faster.

Reply:
what about just having a `HashedSeed` type?

Reply:
The user seed is removed in #60204, without an additional round with `SeedHasher`, by just storing two UInt128 values instead of one. Printing is uglier, but who cares.